### ISSUE TITLE
Failing tests: Invalid boundary in multipart form: b''

### ISSUE THREAD

**Issue by leplatrem** [Kinto/kinto-attachment#169]
```
errors.py                  133 ERROR    Invalid boundary in multipart form: b''
Traceback (most recent call last):
  File "/home/travis/build/Kinto/kinto-attachment/.tox/py36/lib/python3.6/site-packages/pyramid/tweens.py", line 41, in excview_tween
    response = handler(request)
  File "/home/travis/build/Kinto/kinto-attachment/.tox/py36/lib/python3.6/site-packages/kinto/core/events.py", line 160, in tween
    response = handler(request)
  File "/home/travis/build/Kinto/kinto-attachment/.tox/py36/lib/python3.6/site-packages/pyramid/router.py", line 73, in handle_request
    has_listeners and notify(NewRequest(request))
  File "/home/travis/build/Kinto/kinto-attachment/.tox/py36/lib/python3.6/site-packages/pyramid/registry.py", line 109, in notify
    [_ for _ in self.subscribers(events, None)]
  File "/home/travis/build/Kinto/kinto-attachment/.tox/py36/lib/python3.6/site-packages/zope/interface/registry.py", line 442, in subscribers
    return self.adapters.subscribers(objects, provided)
  File "/home/travis/build/Kinto/kinto-attachment/.tox/py36/lib/python3.6/site-packages/zope/interface/adapter.py", line 607, in subscribers
    subscription(*objects)
  File "/home/travis/build/Kinto/kinto-attachment/.tox/py36/lib/python3.6/site-packages/pyramid/config/adapters.py", line 101, in derived_subscriber
    return subscriber(arg[0])
  File "/home/travis/build/Kinto/kinto-attachment/.tox/py36/lib/python3.6/site-packages/cornice/pyramidhook.py", line 152, in wrap_request
    setattr(request, 'errors', Errors(localizer=request.localizer))
  File "/home/travis/build/Kinto/kinto-attachment/.tox/py36/lib/python3.6/site-packages/pyramid/decorator.py", line 43, in __get__
    val = self.wrapped(inst)
  File "/home/travis/build/Kinto/kinto-attachment/.tox/py36/lib/python3.6/site-packages/pyramid/i18n.py", line 393, in localizer
    current_locale_name = self.locale_name
  File "/home/travis/build/Kinto/kinto-attachment/.tox/py36/lib/python3.6/site-packages/pyramid/decorator.py", line 43, in __get__
    val = self.wrapped(inst)
  File "/home/travis/build/Kinto/kinto-attachment/.tox/py36/lib/python3.6/site-packages/pyramid/i18n.py", line 409, in locale_name
    locale_name = negotiate_locale_name(self)
  File "/home/travis/build/Kinto/kinto-attachment/.tox/py36/lib/python3.6/site-packages/pyramid/i18n.py", line 159, in negotiate_locale_name
    locale_name = negotiator(request)
  File "/home/travis/build/Kinto/kinto-attachment/.tox/py36/lib/python3.6/site-packages/pyramid/i18n.py", line 143, in default_locale_negotiator
    locale_name = request.params.get(name)
  File "/home/travis/build/Kinto/kinto-attachment/.tox/py36/lib/python3.6/site-packages/webob/request.py", line 839, in params
    params = NestedMultiDict(self.GET, self.POST)
  File "/home/travis/build/Kinto/kinto-attachment/.tox/py36/lib/python3.6/site-packages/webob/request.py", line 794, in POST
    encoding='utf8')
  File "/opt/python/3.6.3/lib/python3.6/cgi.py", line 559, in __init__
    self.read_multi(environ, keep_blank_values, strict_parsing)
  File "/home/travis/build/Kinto/kinto-attachment/.tox/py36/lib/python3.6/site-packages/webob/compat.py", line 168, in read_multi
    'Invalid boundary in multipart form: %r' % (ib,))
ValueError: Invalid boundary in multipart form: b''
```

**Comment by Natim** [comment]
At first I though it was one of thoose:

```kinto_attachment/tests/test_views_attachment.py:
   65 :         resp = self.upload(files=[
  178 :         resp = self.upload(files=[
  191 :         resp = self.upload(files=[
  252 :         resp = self.upload(files=[(self.file_field, b'', b'--fake--')],
  258 :         resp = self.upload(files=[(self.file_field, b'virus.exe',
  264 :         resp = self.upload(files=[('fichierjoint', b'image.jpg', b'--fake--')],```

But I couldn't see any place where it was wrong.

**Comment by Natim** [comment]
I have the same issue on my code too:

With a test that is supposed to handle it...
```
    def test_profile_view_should_be_accessed_with_a_picture_pic_field(self):
        files = [("WRONG", b"my-report.pdf", b"--binary--")]
        content_type, body = self.app.encode_multipart([], files)
        response = self.app.post(
            "/profile/picture",
            body,
            headers={"Authorization": "Bearer 1234", "Content-Type": content_type},
            status=400,
        )
        self.assertFormattedError(
            response,
            400,
            ERRORS.INVALID_POSTED_DATA,
            "Bad Request",
            "Profile picture is missing.",
        )
```

**Comment by Natim** [comment]
Reverting to `cornice < 3.5` fixed it for me.

**Comment by Natim** [comment]
Fixed with https://github.com/Cornices/cornice/pull/506

**Comment by leplatrem** [comment]
Thanks for the investigation!

**Comment by leplatrem** [comment]
Fixed with #162
